### PR TITLE
ColorlessDamageSource as StaticAbility

### LINKED ISSUE #5994: Colorless Damage Source as StaticAbility
- [ ] Remove hidden Keyword `Colorless Damage Source`, it is only used for `Ghostly Flame`
- [ ] update `nonRed` in the Valid Properties to `!Red` for easier checks? `nonred` should still exist in Descriptions

### PR DESCRIPTION
**Pull request by Hanmac**
Closes #5994 

imo the "non" negations should be replaced with "!"

### CODE REVIEW

**Review by tool4ever — DISMISSED** [review]

File: forge-game/src/main/java/forge/game/ForgeScript.java [hunk 1]
```diff
@@ -74,43 +72,39 @@ public static boolean cardStateHasProperty(CardState cardState, String property,
...
-            if (property.endsWith("Source") && isColorlessSource)
-                return false;
-            return source.hasChosenColor() && colors.hasAnyColor(MagicColor.fromName(source.getChosenColor()));
+            boolean non = property.startsWith("non");
+            return !non && source.hasChosenColor() && colors.hasAnyColor(MagicColor.fromName(source.getChosenColor()));
         } else if (property.startsWith("AnyChosenColor")) {
-            if (property.endsWith("Source") && isColorlessSource)
-                return false;
-            return source.hasChosenColor()
+            boolean non = property.startsWith("non");
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ForgeScript.java:84]: this didn't even support non before, can be cleaned up

File: forge-game/src/main/java/forge/game/ForgeScript.java [hunk 2]
```diff
@@ -74,43 +72,39 @@ public static boolean cardStateHasProperty(CardState cardState, String property,
...
-            if (property.endsWith("Source") && isColorlessSource)
-                return false;
-            return property.startsWith("non") != colors.isAllColors();
+            boolean non = property.startsWith("non");
+            return non != colors.isAllColors();
         } else if (property.contains("MonoColor")) { // ... Card is monocolored
-            if (property.endsWith("Source") && isColorlessSource)
-                return false;
-            return property.startsWith("non") != colors.isMonoColor();
+            boolean non = property.startsWith("non");
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ForgeScript.java:78]: no script with nonMonoColor exists, can already be cleaned up

File: forge-game/src/main/java/forge/game/ForgeScript.java [hunk 3]
```diff
@@ -33,36 +35,32 @@ public class ForgeScript {
...
-            if (non && withSource && isColorlessSource) {
-                return false;
-            }
             return non != colors.isColorless();
         } else if (property.contains("MultiColor")) {
             // ... Card is multicolored
-            if (property.endsWith("Source") && isColorlessSource)
-                return false;
-            return property.startsWith("non") != colors.isMulticolor();
+            boolean non = property.startsWith("non");
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ForgeScript.java:61]: no script with nonMultiColor exists, can already be cleaned up

**Comment by Hanmac** [comment]
@tool4ever probably remove them from this too?

https://github.com/Card-Forge/forge/blob/f0f6e54a0fd90034095adc3721edd3be14bbc020/forge-gui/src/main/java/forge/gui/card/CardScriptParser.java#L443-L444

**Comment by tool4ever** [comment]
yea, either now or I do it later

**Review by tool4ever — APPROVED** [review]

**Pull request merged**